patches/server/0938-Fix-certain-inventories-returning-null-Locations.patch

```diff
@@ -3,47 +3,48 @@ From: Jake Potrebic <jake.m.potrebic@gmail.com>
-         SimpleContainer inventorysubcontainer = this.inventory;
+diff --git a/src/main/java/net/minecraft/world/SimpleContainer.java b/src/main/java/net/minecraft/world/SimpleContainer.java
+index e938e46a142f3764f8224a1fd035e7d541b59c56..9d1ee40456a8d7001eee654a62e62cab2626305a 100644
+--- a/src/main/java/net/minecraft/world/SimpleContainer.java
++++ b/src/main/java/net/minecraft/world/SimpleContainer.java
+@@ -63,6 +63,16 @@ public class SimpleContainer implements Container, StackedContentsCompatible {
  
--        this.inventory = new SimpleContainer(this.getInventorySize(), (org.bukkit.entity.AbstractHorse) this.getBukkitEntity()); // CraftBukkit
+     @Override
+     public Location getLocation() {
 +        // Paper start
-+        this.inventory = new SimpleContainer(this.getInventorySize(), (org.bukkit.entity.AbstractHorse) this.getBukkitEntity()) // CraftBukkit
-+        {
-+            @Override
-+            public org.bukkit.Location getLocation() {
-+                return AbstractHorse.this.getBukkitEntity().getLocation();
-+            }
-+        };
++        // When the block inventory does not have a tile state that implements getLocation, e. g. composters
++        if (this.bukkitOwner instanceof org.bukkit.inventory.BlockInventoryHolder blockInventoryHolder) {
++            return blockInventoryHolder.getBlock().getLocation();
++        }
++        // When the bukkit owner is a bukkit entity, but does not implement Container itself, e. g. horses
++        if (this.bukkitOwner instanceof org.bukkit.entity.Entity entity) {
++            return entity.getLocation();
++        }
 +        // Paper end
-         if (inventorysubcontainer != null) {
-             inventorysubcontainer.removeListener(this);
-             int i = Math.min(inventorysubcontainer.getContainerSize(), this.inventory.getContainerSize());
+         return null;
+     }
+ 
 diff --git a/src/main/java/net/minecraft/world/inventory/BeaconMenu.java b/src/main/java/net/minecraft/world/inventory/BeaconMenu.java
-index 335d0a85378c04dd466fafd42048b2474c815cb9..fb1d71143a0344432af9dc5c3085e217a2778f10 100644
+index 335d0a85378c04dd466fafd42048b2474c815cb9..50ddd689ece23dba50b8c4a911700bd4b3391cb1 100644
 --- a/src/main/java/net/minecraft/world/inventory/BeaconMenu.java
 +++ b/src/main/java/net/minecraft/world/inventory/BeaconMenu.java
-@@ -49,6 +49,12 @@ public class BeaconMenu extends AbstractContainerMenu {
-             public int getMaxStackSize() {
-                 return 1;
-             }
+@@ -40,6 +40,13 @@ public class BeaconMenu extends AbstractContainerMenu {
+         super(MenuType.BEACON, syncId);
+         this.player = (Inventory) inventory; // CraftBukkit - TODO: check this
+         this.beacon = new SimpleContainer(1) {
 +            // Paper start
-+            @Override
-+            public org.bukkit.Location getLocation() {
-+                return context.getLocation();
++            {
++                if (context.getLocation() != null) {
```

Comment:
I don't know if I should check the context like this or with `context != ContainerLevelAccess.NULL`, although this would permit for other implementations that would return null

Reply:
Don't do beacon like this. The correct solution is to add a block inventory holder for beacons, which is done here https://github.com/PaperMC/Paper/pull/7400

Reply:
Reverted that last change again. I was thinking about doing it like that, but that was way more complicated than I thought and felt like the beacon was a bit more comparable to the composter.

But otherwise is this ok?